metro/producer/v1/producer_api.proto

```diff
@@ -0,0 +1,65 @@
+// Protocol Buffers, (c) Razorpay 2020
+
+syntax = "proto3";
+
+package rzp.metro.producer.v1;
+
+option csharp_namespace = "Rzp.Metro.Producer.V1";
+option go_package = "metro/producer/v1;producerv1";
+option java_multiple_files = true;
+option java_outer_classname = "ProducerApiProto";
+option java_package = "com.rzp.metro.producer.v1";
+option objc_class_prefix = "RMP";
+option php_namespace = "Rzp\\Metro\\Producer\\V1";
+
+
+service ProducerApi {
+
+    rpc Produce(ProduceRequest) returns (ProduceResponse);
+
+}
+
+message ProduceRequest {
+
+  string topic = 1;
+
+  repeated Message messages = 2;
+}
+
+
+message ProduceResponse {
+  // The server-assigned ID of each Produceed message, in the same order as
+  // the messages in the request. IDs are guaranteed to be unique within
+  // the topic.
+  repeated string message_ids = 1;
+}
+
+message Message {
+  // The message data field. If this field is empty, the message must contain
+  // at least one attribute.
+  bytes data = 1;
+
+  // Attributes for this message. If this field is empty, the message must
+  // contain non-empty data. This can be used to filter messages on the
+  // subscription.
+  map<string, string> attributes = 2;
+
+  // ID of this message, assigned by the server when the message is Produceed.
+  // Guaranteed to be unique within the topic. This value may be read by a
+  // subscriber that receives a `PubsubMessage` via a `Pull` call or a push
+  // delivery. It must not be populated by the Produceer in a `Produce` call.
+  string message_id = 3;
```

Comment:
we would want the client to generate the message_id, and not really worry about dedup at metro